src/util/field/field.ts

```diff
@@ -0,0 +1,37 @@
+import * as O from 'fp-ts/Option'
+import { pipe } from 'fp-ts/function'
+import { escapeRegExp } from '@/util/util'
+
+/**
+ * Returns a function that checks if a given string s is a valid number field format.
```

Comment:
Here, I wouldn't call it a "field". The purpose of the function is to check that a given string is a valid number (with a few assumptions), whatever its use case.